src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonerIepLevelReviewService.kt

```diff
@@ -269,6 +269,34 @@ class PrisonerIepLevelReviewService(
       }
     }
 
+  @Transactional
+  suspend fun processPrisonerUpdatedEvent(prisonOffenderEvent: HMPPSDomainEvent) {
+    val alertsChanged: Boolean = prisonOffenderEvent.additionalInformation.categoriesChanged?.contains("ALERTS") ?: false
+
+    if (alertsChanged) {
+      updateNextReviewDate(prisonOffenderEvent)
+    } else {
+      log.debug("Ignoring 'prisoner-offender-search.prisoner.updated' event, alerts didn't change: categoriesChanged = ${prisonOffenderEvent.additionalInformation.categoriesChanged}")
+    }
+  }
+
+  private suspend fun updateNextReviewDate(prisonOffenderEvent: HMPPSDomainEvent) {
+    prisonOffenderEvent.additionalInformation.nomsNumber?.let { prisonerNumber ->
+      val offender = offenderSearchService.getOffender(prisonerNumber)
+      nextReviewDateUpdaterService.updateMany(listOf(offender))
+
+      snsService.publishDomainEvent(
+        id = offender.bookingId,
+        nomsNumber = offender.prisonerNumber,
+        occurredAt = LocalDateTime.now(clock),
+        eventType = IncentivesDomainEventType.PRISONER_NEXT_REVIEW_DATE_UPDATED,
+        description = "A prisoner next review date was updated",
+      )
+    } ?: run {
+      log.warn("prisonerNumber null for prisonOffenderEvent: $prisonOffenderEvent ")
```

Comment:
You could additionally mention the result of it being null? For example `prisonerNumber null for prisonOffenderEvent: $prisonOffenderEvent so could not update next review date`

Reply:
Good idea, I'll add the reason to the warning 👍